cli/fmt.rs

```diff
@@ -32,38 +34,45 @@ fn get_config() -> dprint::configuration::Configuration {
   ConfigurationBuilder::new().deno().build()
 }
 
-fn check_source_files(
+async fn check_source_files(
   config: dprint::configuration::Configuration,
   paths: Vec<PathBuf>,
 ) -> Result<(), ErrBox> {
-  let mut not_formatted_files = vec![];
-  let formatter = dprint::Formatter::new(config);
-
-  for file_path in paths {
-    let file_path_str = file_path.to_string_lossy();
-    let file_contents = fs::read_to_string(&file_path)?;
-    let r = formatter.format_text(&file_path_str, &file_contents);
-    match r {
-      Ok(formatted_text) => {
-        if formatted_text != file_contents {
-          not_formatted_files.push(file_path);
+  let not_formatted_files_count = Arc::new(AtomicUsize::new(0));
+  let formatter = Arc::new(dprint::Formatter::new(config));
+
+  run_parallelized(paths, {
+    let not_formatted_files_count = not_formatted_files_count.clone();
+    move |file_path| {
+      let file_path_str = file_path.to_string_lossy();
+      let file_contents = fs::read_to_string(&file_path)?;
+      let r = formatter.format_text(&file_path_str, &file_contents);
+      match r {
+        Ok(formatted_text) => {
+          if formatted_text != file_contents {
+            not_formatted_files_count.fetch_add(1, Ordering::SeqCst);
+          }
+        }
+        Err(e) => {
+          eprintln!("Error checking: {}", &file_path_str);
+          eprintln!("   {}", e);
```

Comment:
This should be a single print statement to prevent a print on another thread to not end up between these.